### ISSUE TITLE
Unable to get completions for user installed packages

### ISSUE THREAD

**Issue by ZacLN** [JuliaEditorSupport/julia-vscode#35]
Maybe we need a configuration option to allow `JULIA_PKGDIR` to be directed at the user's home package directory?

**Comment by davidanthoff** [comment]
This is a really difficult problem... 

In general, any package from the user package directory can be used, because the user package directory is already added to ``LOAD_PATH``. But we would have to actually import these packages so that they show up for completion etc. And I'm really not sure we should do that, there are lots of downsides of executing user code in the language server process. The same is true for any user code that is actually edited in VS Code: for any of that to show up in e.g. the current completion list, we would have to execute that user code in the language server process.

I would much prefer a more static analysis tool a la C#'s Roslyn. But that of course is a gigantic project.

So long story short, I'm not sure how to solve this. I think maybe for now we just try to get a new version that works for everything in base out, and then try to solve this later?

We should probably also take a look at Juno and see how they deal with this problem.

**Comment by ZacLN** [comment]
Is a julia -e 
```
include("dependencies.jl")
installstuff()
``` 
in the scripts:compile/install portion of package.json a possibile solution?

**Comment by davidanthoff** [comment]
@ZacLN I think we can close this issue, right? Let me know if not.
